rclcpp/include/rclcpp/experimental/subscription_intra_process.hpp

```diff
@@ -102,6 +102,12 @@ class SubscriptionIntraProcess : public SubscriptionIntraProcessBase
 #endif
   }
 
+  ~SubscriptionIntraProcess()
+  {
+    rcl_ret_t ret = rcl_guard_condition_fini(&gc_);
+    (void)ret;
```

Comment:
I think it would be better to throw exception or at least print error via `RCUTILS_LOG_ERROR_NAMED`, instead of ignoring return code.

```suggestion
    if (RCL_RET_OK != ret) {
      rclcpp::exceptions::throw_from_rcl_error(ret, "failed to finalize guard condition");
    }
```

Reply:
I prefer to use `RCLCPP_ERROR` to print error log here, which avoids throwing an exception on destructor.

Reply:
If throw exception，there is warning "note: in C++11 destructors default to noexcept".   
So I will add output error log.